lib/github_service/commands/add_label.rb

```diff
@@ -30,6 +31,37 @@ def validate_labels(label_names)
         # Then see if any are *still* invalid and split the list
         label_names.partition { |l| GithubService.valid_label?(issue.fq_repo_name, l) }
       end
+
+      def process_extracted_labels(valid_labels, invalid_labels)
+        labels = GithubService.labels(issue.fq_repo_name)
+        invalid_labels.reject! do |label|
+          corrections = DidYouMean::SpellChecker.new(:dictionary => labels).correct(label)
+
+          if corrections.count == 1
+            valid_labels << corrections.first
+          end
+        end
+
+        [valid_labels, invalid_labels]
+      end
+
+      def invalid_label_message(issuer, invalid_labels)
+        message  = "@#{issuer} "
+        message << "Cannot apply the following label"
+        message << "s" if invalid_labels.length > 1
+        message << " because they are not recognized:\n"
+
+        labels = GithubService.labels(issue.fq_repo_name)
+        invalid_labels.each do |bad_label|
+          corrections   = DidYouMean::SpellChecker.new(:dictionary => labels).correct(bad_label)
+          possibilities = corrections.map { |l| "`#{l}`" }.join(", ")
+
+          message << "* `#{bad_label}` "
+          message << "(Did you mean? #{possibilities})" if corrections.any?
+          message << "\n"
+        end
+        message << "\nAll labels for `#{issue.fq_repo_name}`:  https://github.com/#{issue.fq_repo_name}/labels"
```

Comment:
Do you have a screenshot on what all this looks like?  I'm having trouble visualizing.

Reply:
Yeah, I will try running a bit today locally, but I ran out of API calls when testing it last week... 

https://github.com/GrumpyTech/manageiq-ui-classic/pull/1#issuecomment-578345729

I was starting and stopping `sidekiq` a bunch so that was breaking the cache everytime, so I used up my API limit very quickly.

I will run it again this morning and get you a screenshot.

Reply:
Examples can now be found here:

https://github.com/GrumpyTech/manageiq-ui-classic/pull/1#issuecomment-578347807